services/worker-manager/src/providers/aws.js

```diff
@@ -0,0 +1,99 @@
+const {Provider} = require('./provider');
+const aws = require('aws-sdk');
+
+class AwsProvider extends Provider {
+  constructor({workerPool}) {
+    super({workerPool});
+    this.configSchema = 'config-aws';
+
+    const {region} = workerPool.config;
+    const {apiVersion} = workerPool.config;
+
+    aws.config.update({region});
+    this.iam = new aws.IAM({apiVersion});
+  }
+
+  async setup() {
+    // create IAM user
+    const {region} = this.workerPool.config;
+    const {account} = this.workerPool.providerData;
+    const userName = `wm-aws-provider-${this.workerPool.providerData.userName}`;
+
+    await readModifySet({
+      read: async () => await this.iam.getUser({UserName: userName}),
+      set: async () => await this.iam.createUser({UserName: userName}),
+    });
+
+    // create IAM policy
+    const ec2policy = {
+      "Statement": [{
+        "Effect": "Allow",
+        "Action": [
+          "ec2:DescribeAvailabilityZones",
+          "ec2:DescribeInstances",
+          "ec2:TerminateInstances",
+          "ec2:RunInstances",
+        ],
+        "Resource": `arn:aws:ec2:${region}:${account}:*`,
+      }],
+    };
+    const policyName = 'wm-provider-ec2policy';
+    await this.iam.createPolicy({
+      PolicyDocument: JSON.stringify(ec2policy),
+      PolicyName: policyName,
+    });
+
+    // attach the policy to the user
+    await this.iam.attachUserPolicy({
+      PolicyArn: `arn:aws:iam::aws:policy/${policyName}`,
+      UserName: userName,
+    });
+  }
+}
+
+// this util copy-pasted from gcp provider
```

Comment:
If this proves useful as-is for aws, let's do that for sure!

Reply:
it does look like a nice generic util to me. But I'll wait till this provider is finished and see